projects/project6/src/main/scala/axiom_extraction/Main.scala

```diff
@@ -0,0 +1,138 @@
+package axiom_extraction
+
+
+import net.sansa_stack.rdf.spark.io._
+import org.apache.spark.sql.SparkSession
+import org.apache.jena.riot.Lang
+import org.apache.spark.rdd._
+import org.apache.spark.sql.{DataFrame, SQLContext}
+import org.apache.spark.{SparkConf, SparkContext}
+import scala.collection.mutable.ListBuffer
+import org.apache.spark.sql.functions._
+import org.apache.spark.ml.fpm.FPGrowth
+
+
+object Main extends App {
+
+  var DEBUG: Boolean = false
+  
+  override def main(args: Array[String]) = {
+    val spark = SparkSession.builder
+      .appName(s"AxiomExtraction")
+      .master("local[*]")
+      .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
+      .getOrCreate()
+    val resource = "test_input.nt"
+    val path = getClass.getResource(resource).getPath()
+    val df: DataFrame = spark.read.rdf(Lang.NTRIPLES)(path)
+    Console.println("===========Input DataFrames===========")
+    df.show()
+    
+    Console.println("===========Predicates===========")
+    //selecting all distinct predicates
+    val distinctValuesDF = df.select("p").distinct
+    //selecting all distinct instances
+    val distinctSubjectsDF = df.select("s").distinct
+    val distinctObjectsDF = df.select("o").distinct
+    distinctValuesDF.show()
+    
+    //all distinct instances in a list
+    val instances = distinctSubjectsDF.union(distinctObjectsDF).distinct().rdd.map(r => r(0)).collect()
+    
+    //list of subsamples for instance pair extraction
+    var dfList : ListBuffer[DataFrame] = new ListBuffer()
+    
+    var transactions : ListBuffer [String] = new ListBuffer()
+    
+    
+    Console.println("===========Extendet Predicates===========")
+    //printing out distinct predicate values
+    distinctValuesDF.select("p").take((distinctValuesDF.count()).toInt).foreach(printWithNeg)    
+    
+    //sampling the pair groups
+    instances.foreach {x => getInstanceFilter(x,df,dfList)}
+    
+    Console.println("===========Transactions Extracted===========")
+    //extracting pairs and generating transactions
+    dfList.foreach(x=>transactions += extractTransactions(x))
+    transactions.foreach(println)
+    
+    import spark.implicits._
+    val dataset = spark.createDataset(transactions).map(t => t.split(" ")).toDF("items")

```

Comment:
You can use also a SANSA functionality for converting a DF to a Dataset.
See here(https://github.com/SANSA-Stack/SANSA-RDF/blob/develop/sansa-rdf-spark/src/test/scala/net/sansa_stack/rdf/spark/model/df/DFTripleOpsTests.scala#L31)  as an example of how to do it.

Reply:
Here in fact I use a DataFrame as an input for the FR-Growth, so i just need to mark "items" correspondingly. So I'd rather use this form.

Reply:
Yes, you could keep them as your customized dataset/DF.